includes/shortcodes.php

```diff
@@ -149,123 +149,20 @@ function give_form_shortcode( $atts ) {
-			! $donationFormHasSession
-		) {
-			unset( $query_string['giveDonationAction'] );
-			$hasAction    = false;
-			$isAutoScroll = 0;
-		}
-
-		// Build iframe url.
-		$url = Give()->routeForm->getURL( get_post_field( 'post_name', $formId ) );
-
-		if ( ( $hasAction && 'showReceipt' === $query_string['giveDonationAction'] ) || FormUtils::isViewingFormReceipt() ) {
-			$url = FormUtils::getSuccessPageURL();
-
-		} elseif ( ( $hasAction && 'failedDonation' === $query_string['giveDonationAction'] ) ) {
-			$url                                     = Give()->templates->getTemplate( $activeTheme )->getFailedPageURL( $formId );
-			$query_string['showFailedDonationError'] = 1;
-		}
-
-		$iframe_url = add_query_arg(
-			array_merge( [ 'giveDonationFormInIframe' => 1 ], $query_string ),
-			trailingslashit( $url )
-		);
-
-		$iframe_url = GlobalUtils::removeDonationAction( $iframe_url );
-
-		$uniqueId         = uniqid( 'give-' );
-		$buttonModeActive = 'button' === $atts['display_style'];
-
-		ob_start();
-		include Give()->templates->getTemplate( $activeTheme )->getLoadingView();
-
-		$loader = sprintf(
-			'<div class="iframe-loader">%1$s</div>',
-			ob_get_clean()
-		);
-
-		// Set iframe.
-		$iframe = sprintf(
-			'<iframe
-						name="give-embed-form"
-						src="%1$s"
-						data-autoScroll="%2$s"
-						onload="Give.initializeIframeResize(this)"
-						style="border: 0; visibility: hidden; min-height: %3$spx;"></iframe>%4$s',
-			$iframe_url,
-			$isAutoScroll,
-			$formStartingHeight,
-			$loader
-		);
-
-		// Show button in button mode and hide Iframe.
-		if ( $buttonModeActive ) {
-			printf(
-				'<div class="js-embed-form-modal-launcher-wrap">
-						<button
-						type="button"
-						class="js-give-embed-form-modal-opener"
-						data-form-id="%1$s"%3$s>%2$s</button>
-					</div>',
-				$uniqueId,
-				! empty( $atts['continue_button_title'] ) ? $atts['continue_button_title'] : __( 'Click to donate', 'give' ),
-				! empty( $atts['button_color'] ) ? " style=\"background-color: {$atts['button_color']}\"" : ''
-			);
-
-			// Insert iframe inside modal HTML.
-			$iframe = sprintf(
-				'
-						<div class="modal-inner-wrap">
-							<div class="modal-content">
-								<a href="#" class="close-btn js-give-embed-form-modal-closer" aria-label="%3$s" data-form-id="%4$s" rel="nofollow">%7$s<span>&times;</span></a>
-								<iframe
-									name="give-embed-form"
-									class="in-modal"
-									data-src="%1$s"
-									data-autoScroll="%2$s"
-									style="border: 0; visibility: hidden; min-height: %5$spx;"></iframe>
-									%6$s
-							</div>
-						</div>
-						',
-				$iframe_url,
-				$isAutoScroll,
-				__( 'Close modal', 'give' ),
-				$uniqueId,
-				$formStartingHeight,
-				$loader,
-				__( 'CLOSE', 'give' )
-			);
-		}
-
-		printf(
-			'<div class="give-embed-form-wrapper%1$s" id="%2$s">%3$s</div>',
-			$buttonModeActive ? ' is-hide' : '',
-			$uniqueId,
-			$iframe
-		);
-
+		echo $iframeView->setFormId( $formId )
```

Comment:
Let's set this all to a variable and then echo that out, for ease of debugging with Xdebug or similar.

Reply:
It's a good thought, but I think this is fine. When folks are debugging they're often tweaking code momentarily, anyway. What debugs easily and what's clear and concise aren't always the same, and I'd rather lean towards the latter in terms of what we commit.